commons/src/main/java/org/infinispan/counter/api/StrongCounter.java

```diff
@@ -0,0 +1,94 @@
+package org.infinispan.counter.api;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The strong consistent counter interface.
+ * <p>
+ * It provides atomic updates for the counter. All the operations are perform asynchronously and they complete the
+ * {@link CompletableFuture} when completed.
+ * <p>
+ * The implementation may support weakly consistent reads via {@link #weakGetValue()}.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public interface StrongCounter {
+
+   /**
+    * @return The counter name.
+    */
+   String getName();
+
+
+   /**
+    * It fetches the current value.
+    * <p>
+    * It may go remotely to fetch the current value.
+    *
+    * @return The current value.
+    */
+   CompletableFuture<Long> getValue();
+
+   /**
+    * Returns the locally cached value.
+    * <p>
+    * This value may not up-to-date and it may no have the most recent updates from local and remote nodes.
+    *
+    * @return The locally cached value.
+    */
+   default long weakGetValue() {
```

Comment:
initially, my idea was to have a "L1" cache with the counter value. It uses the cluster listeners to update it so it may miss some local updates. However, it gives me more troubles and I'm no longer sure if it is useful to keep it around. Do you think I should remove it from Strong and Weak counters?